File: diary-server/src/main/java/com/diary/server/web/dto/DiaryResponse.java
package com.diary.server.web.dto;

import com.diary.server.domain.Diary;

import java.time.Instant;
import java.time.LocalDate;

public record DiaryResponse(
        Long id,
        LocalDate date,
        String content,
        String location,
        Double latitude,
        Double longitude,
        Instant createdAt,
        Instant updatedAt
) {
    public static DiaryResponse from(Diary diary) {
        return new DiaryResponse(
                diary.getId(),
                diary.getDiaryDate(),
                diary.getContent(),
                diary.getLocation(),
                diary.getLatitude(),
                diary.getLongitude(),
                diary.getCreatedAt(),
                diary.getUpdatedAt()
        );
    }
}

File: diary-server/src/main/kotlin/com/diary/server/web/dto/DiaryResponse.kt
package com.diary.server.web.dto

import com.diary.server.domain.Diary
import java.time.Instant
import java.time.LocalDate

data class DiaryResponse(
    val id: Long?,
    val date: LocalDate?,
    val content: String,
    val location: String?,
    val latitude: Double?,
    val longitude: Double?,
    val createdAt: Instant?,
    val updatedAt: Instant?
) {
    companion object {
        fun from(diary: Diary): DiaryResponse = DiaryResponse(
            id = diary.id,
            date = diary.diaryDate,
            content = diary.content,
            location = diary.location,
            latitude = diary.latitude,
            longitude = diary.longitude,
            createdAt = diary.createdAt,
            updatedAt = diary.updatedAt
        )
    }
}
